src/Microsoft.SqlTools.ServiceLayer/AzureBlob/SharedAccessSignatureCreator.cs

```diff
@@ -0,0 +1,117 @@
+﻿//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+using System;
+using Microsoft.SqlServer.Management.Smo;
+using Azure.Storage.Blobs;
+using Azure.Storage;
+using Azure.Storage.Sas;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlTools.ServiceLayer.AzureBlob.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureBlob
+{
+    class SharedAccessSignatureCreator
+    {
+
+
+        private Server sqlServer;
+
+        public SharedAccessSignatureCreator(Server sqlServer)
+        {
+            this.sqlServer = sqlServer;
+        }
+
+        public string CreateSqlSASCredential(string accountName, string accountKey, string containerUri, string expirationDateString)
+        {
+            DateTimeOffset? expirationDate = null;
+            if (!String.IsNullOrEmpty(expirationDateString))
+            {
+                expirationDate = DateTimeOffset.Parse(expirationDateString);
+            }
+            var containerClient = new BlobContainerClient(new Uri(containerUri), new StorageSharedKeyCredential(accountName, accountKey));
+            Uri secretStringUri = GetServiceSasUriForContainer(containerClient, null, expirationDate);
+            string secretString = secretStringUri.ToString().Split('?')[1];
+            string identity = "Shared Access Signature";
```

Comment:
do you need to make sure this id is not already been used?

Reply:
This is not unique id. Sql Server can have stored multiple shared access signatures for blob containers (one per blob container). You can read more [here](https://docs.microsoft.com/en-us/sql/t-sql/statements/create-credential-transact-sql?view=sql-server-ver15#:~:text=IDENTITY%20%3D%27identity_name%27%20Specifies%20the%20name%20of%20the%20account,signature%20%28SAS%29%2C%20the%20IDENTITY%20is%20SHARED%20ACCESS%20SIGNATURE.)

Reply:
This identity is not unique id. Sql Server can store more than one shared access signature (one per blob container). You can read more [here](https://docs.microsoft.com/en-us/sql/t-sql/statements/create-credential-transact-sql?view=sql-server-ver15#:~:text=IDENTITY%20%3D%27identity_name%27%20Specifies%20the%20name%20of%20the%20account,signature%20%28SAS%29%2C%20the%20IDENTITY%20is%20SHARED%20ACCESS%20SIGNATURE.)